packages/slate-lists/src/commands/toggle-list.js

```diff
@@ -11,6 +11,9 @@ export default ({ blocks }, editor, options = {}) => {
   if (sameType) {
     editor.unwrapList();
   } else {
-    editor.setNodeByKey(list.key, type);
+    editor.setNodeByKey(
+      list.key,
+      Object.assign({}, options.nodeProperties, { type })
```

Comment:
Why `options.nodeProperties` here? What do you think of something like this:

```
editor.setNodeByKey(list.key, { type, data: options.data || {} })
```

Reply:
I was just allowing people to set all properties on the node, but I don't have any issue limiting it do the node's `data` property. Will update

Reply:
Ah, gotcha. I'd prefer to stick with data right now, but open to something different if there's a good use case.